src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java

```diff
@@ -56,7 +58,12 @@ public DefaultMonitorService(Log logger) {
             propertySet.getIntegerProperty(PropertyKey.monitorDisposalTime).getValue(),
             monitorService,
             logger),
-        Executors::newCachedThreadPool,
+        () -> Executors.newCachedThreadPool(r -> {
+          final Thread monitoringThread = new Thread(r);
+          monitoringThread.setDaemon(true);
+          monitoringThread.setPriority(Thread.NORM_PRIORITY);
```

Comment:
Do we need this line? It'll be normal by default already no?

Reply:
Not necessarily, it depends on the parent thread.
So if users decide to run the driver in a separate thread with priority 2, the monitoring thread will inherit that priority.

Although in this case I suppose we want the monitoring thread to have the same priority as the parent thread? @sergiyv-bitquill

Reply:
Yes, I agree. We want monitoring threads run with the same priority as main one. We don't want to steal cpu cycles from main app.

Reply:
Okay, I will remove this line.